components/collective-page/sections/SponsorsAndParticipants.js

```diff
@@ -68,6 +70,17 @@ const Participants = ({ collective: event, LoggedInUser }) => {
 
   const canEditEvent = LoggedInUser && LoggedInUser.canEditEvent(event);
 
+  React.useEffect(() => {
+    const refreshData = async () => {
+      if (canEditEvent) {
+        await refetch();
+        setIsRefetched(true);
+      }
+    };
+
+    refreshData();
+  }, [LoggedInUser]);
```

Comment:
Looks like this can be simplified

```suggestion
  React.useEffect(async () => {
    if (canEditEvent) {
      await refetch();
      setIsRefetched(true);
    }
  }, [LoggedInUser]);
```

Reply:
AFIAK useEffect doesn't directly allow an async function as its callback. It actually recommends this way of doing it. https://github.com/facebook/react/issues/14326

Reply:
Ah yes, useEffect's return must be a function or void right? Then it's all agood

Reply:
Alternative is to use a promise, but at this point it really goes down to personal preferences. No need to make the change

```suggestion
  React.useEffect(() => {
    if (canEditEvent) {
      refetch().then(() => setIsRefetched(true));
    }
  }, [LoggedInUser]);
```

Reply:
> Ah yes, useEffect's return must be a function or void right? Then it's all good

@Betree Yes, the callback should return a cleanup function or void. Should I refactor it to promises or are we good?

Reply:
We're good :slightly_smiling_face: